services/api/src/resources/environment/resolvers.ts

```diff
@@ -308,6 +309,68 @@ export const getEnvironmentByKubernetesNamespaceName: ResolverFn = async (
     ctx
   );
 
+  export const getEnvironmentsByKubernetes: ResolverFn = async (
+    _,
+    { kubernetes, order, createdAfter },
+    { sqlClientPool, hasPermission, models, keycloakGrant }
+  ) => {
+    const openshift = await openshiftHelpers(
+      sqlClientPool
+    ).getOpenshiftByOpenshiftInput(kubernetes);
+
+    let userProjectIds: number[];
+    try {
+      await hasPermission('project', 'viewAll');
```

Comment:
You only need to do the `openshift:viewAll` permission check here, but that means the query will only work for legacy tokens until #3437 lands. But this is the same behaviour that `allEnvironments` query has with `environment:viewAll` which is also resolved by the same PR.